### PR TITLE
Implement matching service

### PR DESCRIPTION
**Pull request by GenFusion122**
Implementation of the backend for the matching-service that allows users to queue and be matched based on `difficulty` and `language`.

`enqueue` - enqueues the user making the request.
`dequeue` - removes the user from the queue.
`getLength` - retrieves the length of the queue, mainly for supporting purposes.

### CODE REVIEW

**Review by chia-yh — APPROVED** [review]
LGTM, but might be some disjoint between behavior and name of `enqueue` and `dequeue`, thought its just a nit

File: backend/matching-service/controllers/matching.controller.js [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+        res.status(500).send({
+            message: error.message || 'Error adding match to queue.'
+        });
+    }
+
+};
+
+exports.dequeue = (req, res) => {
+    try {
+        let existingMatch = mQueue.checkUser(match);
```
Comment (cjyothika) [inline backend/matching-service/controllers/matching.controller.js]: Can you point me to where match is defined in this method?

Comment (cjyothika) [reply backend/matching-service/controllers/matching.controller.js]: Also, maybe the type of match should be checked?

**Review by cjyothika — COMMENTED** [review]
Generally, it looks good but I have a few questions.

File: backend/matching-service/controllers/matching.controller.js (hunk 1 above)
Comment (GenFusion122) [reply backend/matching-service/controllers/matching.controller.js]: Yeah it's not defined here my b, lemme fix it.

**Review by chia-yh — APPROVED** [review]
LGTM

**Review by cjyothika — APPROVED** [review]
LGTM

**Pull request merged**